pylint/checkers/deprecated.py

```diff
@@ -189,6 +204,26 @@ def deprecated_classes(self, module: str) -> Iterable[str]:
         # pylint: disable=unused-argument
         return ()
 
+    def deprecated_attributes(self) -> Container[str]:
+        """Callback returning the deprecated attributes.
+
+        Returns:
+            collections.abc.Container of deprecated attribute names.
+        """
+        return ()
+
+    def check_deprecated_attribute(self, node: astroid.Attribute) -> None:
+        """Checks if the attribute is deprecated."""
+        inferred_expr = safe_infer(node.expr)
+        if not isinstance(inferred_expr, (nodes.ClassDef, nodes.Module)):
+            return
+        attribute_qname = ".".join((inferred_expr.qname(), node.attrname))
+        for deprecated_name in self.deprecated_attributes():
+            if attribute_qname == deprecated_name:
+                self.add_message(
+                    "deprecated-attribute", node=node, args=(node.attrname,)
```

Comment:
This might be more helpful:
```suggestion
                    "deprecated-attribute", node=node, args=(attribute_qname,)
```

Reply:
When I run "python -m pytest unittest_deprecated.py" this command in the unittest_deprecated.py I get an error "FAILED unittest_deprecated.py::TestDeprecatedChecker::test_deprecated_attribute - AttributeError: 'Assign' object has no attribute 'expr'"  can you tell me the cause of this ? As I think if all the test are not passed in this then functional testing will also not work

Reply:
Did you pull the latest commits I pushed?

Reply:
Yes I re cloned it with the latest comits

Reply:
Alright, no problem. I still can't reproduce the error you are seeing, though.

Reply:
I am running the command after making the changes in unittest_depreacted.py without them it works .